ocpmodels/trainers/forces_trainer.py

```diff
@@ -208,14 +208,22 @@ def predict(
                     )
                 ]
                 predictions["id"].extend(systemids)
-                predictions["energy"].extend(
-                    out["energy"].to(torch.float16).tolist()
-                )
                 batch_natoms = torch.cat(
                     [batch.natoms for batch in batch_list]
                 )
                 batch_fixed = torch.cat([batch.fixed for batch in batch_list])
-                forces = out["forces"].cpu().detach().to(torch.float16)
+                # total energy requires predictions are saved in float32
+                # default is ads energy not total energy
+                if self.config["dataset"].get("total_energy", False):
```

Comment:
The OC22 dataloader defaults to total energies even without the `total_energy` flag, right?

If yes, then the only (required) use of the `total_energy` flag is in this block to write float32s. That's a non-obvious dependency.

Instead, let's add a new `prediction_dtype` flag and refactor the surrounding logic to default to float16, but set to float32 in the OC22 configs. Also add it to the `task` subtree please instead of the `dataset`.

Add it to the example yaml files as well please, and document both this new flag, and the OC22 make_submission command in [TRAIN.md](https://github.com/Open-Catalyst-Project/ocp/blob/main/TRAIN.md).

Reply:
I included the `prediction_dtype` flag in this commit https://github.com/Open-Catalyst-Project/ocp/pull/421/commits/2d21fd9c8e1836a28c375963a201d0a7eb11a156, but I still need to update the example yaml and document this flag and the OC22 make_submission command in TRAIN.md.